enhancements/console/workflow-guides.md

```diff
@@ -0,0 +1,175 @@
+* Introduce default storage for Workflow Guides.
+* Introduce mechanism how to connect different parts of OpenShift Console (Operators, Workloads, ...) with the Workflow Guides.
+* **Extenstion:**: Introduce mechanism for defining and displaying list of required resources, that need to be created in order to complete the Workflow Guide.
+
+### User Stories
+
+#### Story 1
+
+As an administrator of an OpenShift cluster, I need a guide to walk me through how to install OpenShift Serverless modules (Serving and Eventing) in a cluster.
+
+#### Story 2
+
+As an administrator of an OpenShift cluster, I need a guide to walk me through how to update an OpenShift cluster.
+
+#### Story 3
+
+As a developer, I need a guide to walk me through how to deploy an existing application as a serverless workload.
+
+#### Story 4
+
+As a operator creator I want to provide operator consumers with a guide on how to install and user the my operator.
+
+#### Story 5
+
+As a OpenShift evangelist I want to link developers on StackOverflow with complete Workflow Guides, for their issues.
+
+### Implementation Details
+
+1. GitHub will be used creating and storing the Workflow Guides. GitHub provides rich markdown options, including images and videos, which will come handy in the process of creating a Workflow Guide. Since we have storage limits in ETCD, it's for the best to use a technology that both storest the content and provides rich markdown. Also by using GitHub we solve the issue with  Workflow Guides versioning.
+2. For linking the OpenShift Console and the Workflow Guides `ConsoleLinks` CRD will be used. To suite the need following extension of the `ConsoleLinks` is needed:
+   1. Add additional `ConsoleLinkLocation` type to the `ConsoleLink` [API](https://github.com/openshift/api/blob/master/console/v1/types_console_link.go#L65-L74) named `WorkflowGuide`.
+   2. If `ConsoleLinkLocation` type is set to `WorkflowGuide`, it will be possible to define a `WorkflowGuideSpec` with following fields:
+      * Section - string - Defines area of interest, e.g. `operators`.
+      * Admin - bool - Defines if the Workflow Guide is cluster admin oriented (user has to have cluster admin permisions).
+      * Version - string - Defines the version of the `WorkflowGuide`
+      * **Extenstion:**: ResourceCheckName - string - Defines the name of the `ResourceCheck` CR.
+
+3. RedHat supported Workflow Guides will be places in new `openshift/workflow-guides` repository. Each root folder in the repository will be named after area of interest (e.g. top items from the left navigation menu -  Operators, Workloads, Serverless, ..., Administration).
+4. The `openshift/workflow-guides` repository will also contain a `manifests/` root folder that will contain all yaml for all the official Workflow Guides `ConsoleLinks` CRs. All the manifests will be copied into the `console-operator` image upon it's build. Alternativly there could be a small `workflow-guides-operator` that will be:
```

Comment:
> All the manifests will be copied into the `console-operator` image upon it's build.

We probably need to vendor them into the repo for repeatable builds.

Reply:
yes will have to.

Or we could have a dedicated operator that would take care of that and also updates.. and possibly other tasks (TBDefined..)